s3/s3_operations.cpp

```diff
@@ -1334,58 +1334,57 @@ namespace irods_s3 {
+        // the objPath and get the fd from that. 
+        irods::file_object_ptr file_obj = boost::dynamic_pointer_cast<irods::file_object>(_ctx.fco());
+        int fd = 0;
+        for (int i = 0; i < NUM_L1_DESC; ++i) {
+            if (L1desc[i].inuseFlag && L1desc[i].dataObjInp && L1desc[i].dataObjInfo) {
+                if (L1desc[i].dataObjInp->objPath == file_obj->logical_path()) {
+                    fd = i;
+                    break;
                 }
             }
-            
-            if (fd_data.exists(fd)) {
-                per_thread_data data = fd_data.get(fd);
-                if (data.dstream_ptr && data.s3_transport_ptr && data.s3_transport_ptr->is_open_cache_file()) {
-
-                    // do a stat on the cache file, populate stat_buf, and return
-                    std::string cache_file_physical_path = data.s3_transport_ptr->get_cache_file_path();
-                         
-                    const int status = stat(cache_file_physical_path.c_str(), _statbuf );
-
-                    // return an error if necessary
-                    if (status < 0) {
-                        const int err_status = UNIX_FILE_STAT_ERR - errno;
-                        return ERROR(err_status, fmt::format(
-                                    "Stat error for \"{}\", errno = \"{}\", status = {}.",
-                                    cache_file_physical_path.c_str(), strerror(errno), err_status));
-                    }
-
-                    auto result = SUCCESS();
-                    result.code(status);
-                    return result;
+        }
+        
+        if (fd_data.exists(fd)) {
+            per_thread_data data = fd_data.get(fd);
+            if (data.dstream_ptr && data.s3_transport_ptr && data.s3_transport_ptr->is_cache_file_open()) {
+
+                // do a stat on the cache file, populate stat_buf, and return
+                std::string cache_file_physical_path = data.s3_transport_ptr->get_cache_file_path();
+                     
+                const int status = stat(cache_file_physical_path.c_str(), _statbuf );
+
+                // return an error if necessary
+                if (status < 0) {
+                    const int err_status = UNIX_FILE_STAT_ERR - errno;
+                    return ERROR(err_status, fmt::format(
+                                "Stat error for \"{}\", errno = \"{}\", status = {}.",
+                                cache_file_physical_path.c_str(), strerror(errno), err_status));
                 }
-            }
 
-            // there is not an open cache file, do the normal HEAD to S3 
-            return s3_file_stat_operation_with_flag_for_retry_on_not_found(_ctx, _statbuf, false);
-        } else {
-            return s3_file_stat_operation_with_flag_for_retry_on_not_found(_ctx, _statbuf, false);
+                auto result = SUCCESS();
+                result.code(status);
+                return result;
```

Comment:
Feels a little dirty to suggest this, but `SUCCESS()` is a shortcut for `CODE(0)`. So...

```suggestion
                return CODE(status);
```

https://github.com/irods/irods/blob/aff93a3fbe417c8ec2e9d834b5eb2d6b2ae31083/lib/core/include/irods/irods_error.hpp#L133-L134

Please ignore if you feel like it. :p

Reply:
I went ahead and made the change.